lib/types/local-time.js

```diff
@@ -27,257 +29,209 @@ const nanoSecInMillis = Long.fromNumber(1000000);
-    if (typeof nanoseconds !== "number") {
-        nanoseconds = 0;
+    set second(newValue) {
+        this.internal.second = newValue;
     }
-    return new LocalTime(
-        Long.fromNumber(milliseconds)
-            .multiply(nanoSecInMillis)
-            .add(Long.fromNumber(nanoseconds)),
-    );
-};
 
-/**
- * Creates a new instance of LocalTime from the bytes representation.
- * @param {Buffer} value
- * @returns {LocalTime}
- */
-LocalTime.fromBuffer = function (value) {
-    if (!(value instanceof Buffer)) {
-        throw new TypeError("Expected Buffer, obtained " + util.inspect(value));
+    get minute() {
+        return this.internal.minute;
     }
-    return new LocalTime(new Long(value.readInt32BE(4), value.readInt32BE(0)));
-};
 
-/**
- * Compares this LocalTime with the given one.
- * @param {LocalTime} other time to compare against.
- * @return {number} 0 if they are the same, 1 if the this is greater, and -1
- * if the given one is greater.
- */
-LocalTime.prototype.compare = function (other) {
-    return this.value.compare(other.value);
-};
+    set minute(newValue) {
+        this.internal.minute = newValue;
+    }
 
-/**
- * Returns true if the value of the LocalTime instance and other are the same
- * @param {LocalTime} other
- * @returns {Boolean}
- */
-LocalTime.prototype.equals = function (other) {
-    return other instanceof LocalTime && this.compare(other) === 0;
-};
+    get hour() {
+        return this.internal.hour;
+    }
 
-/**
- * Gets the total amount of nanoseconds since midnight for this instance.
- * @returns {Long}
- */
-LocalTime.prototype.getTotalNanoseconds = function () {
-    return this.value;
-};
+    set hour(newValue) {
+        this.internal.hour = newValue;
+    }
 
-LocalTime.prototype.inspect = function () {
-    return this.constructor.name + ": " + this.toString();
-};
+    /**
+     * Parses an string representation and returns a new LocalDate.
+     * @param {String} value
+     * @returns {LocalTime}
+     */
+    static fromString(value) {
+        if (typeof value !== "string") {
+            throw new Error("Argument type invalid: " + util.inspect(value));
+        }
 
-/**
- * Returns a big-endian bytes representation of the instance
- * @returns {Buffer}
- */
-LocalTime.prototype.toBuffer = function () {
-    const buffer = utils.allocBufferUnsafe(8);
-    buffer.writeUInt32BE(this.value.getHighBitsUnsigned(), 0);
-    buffer.writeUInt32BE(this.value.getLowBitsUnsigned(), 4);
-    return buffer;
-};
+        const parts = value.split(":");
+        let millis =
+            parseInt(parts[0], 10) * 3600000 + parseInt(parts[1], 10) * 60000;
+        let nanos;
+        if (parts.length === 3) {
+            const secParts = parts[2].split(".");
+            millis += parseInt(secParts[0], 10) * 1000;
+            if (secParts.length === 2) {
+                nanos = secParts[1];
+                //add zeros at the end
```

Comment:
Do we want to have blank space after ``//``? And if so is it possible to check it in prietier?

Reply:
We do. Dunno about prettier.

Reply:
I created issue for this and assigned to me. I will check if it is possible.